test/controllers/util/FormFieldSpec.scala

```diff
@@ -117,140 +117,174 @@ class FormFieldSpec extends AnyWordSpec with MockitoSugar with BeforeAndAfterEac
       // We must only show the user one error at a time so if there are 2 or more, we must show the foremost error
 
-      DateField.validate("1", "-1", "notYear") shouldBe Some("Month must not be a negative number.")
-      DateField.validate("12", "13", "-2022") shouldBe Some("13 is an invalid Month number.")
-      DateField.validate("25", "", "19904") shouldBe Some("There was no number entered for the Month.")
-      DateField.validate("8", "month", "42") shouldBe Some("Month entered must be a whole number.")
+      DateField.validate("1", "-1", "notYear", dateField) shouldBe Some("Month must not be a negative number.")
+      DateField.validate("12", "13", "-2022", dateField) shouldBe Some("13 is an invalid Month number.")
+      DateField.validate("25", "", "19904", dateField) shouldBe Some("There was no number entered for the Month.")
+      DateField.validate("8", "month", "42", dateField) shouldBe Some("Month entered must be a whole number.")
     }
 
     "validate should return only the error for the Year if the Day and Month have no errors" in {
       // We must only show the user one error at a time so if there are 2 or more, we must show the foremost error
 
-      DateField.validate("1", "6", "notYear") shouldBe Some("Year entered must be a whole number.")
-      DateField.validate("12", "1", "-2022") shouldBe Some("Year must not be a negative number.")
-      DateField.validate("25", "8", "19904") shouldBe Some("The year should be 4 digits in length.")
-      DateField.validate("8", "2", "42") shouldBe Some("The year should be 4 digits in length.")
+      DateField.validate("1", "6", "notYear", dateField) shouldBe Some("Year entered must be a whole number.")
+      DateField.validate("12", "1", "-2022", dateField) shouldBe Some("Year must not be a negative number.")
+      DateField.validate("25", "8", "19904", dateField) shouldBe Some("The year should be 4 digits in length.")
+      DateField.validate("8", "2", "42", dateField) shouldBe Some("The year should be 4 digits in length.")
     }
 
     "monthsWithLessThan31Days should only contain months with less than 31 days" in {
       // This is here to ensure that monthsWithLessThan31Days does not accidentally change
 
       DateField.monthsWithLessThan31Days should equal(monthsWithLessThan31Days)
     }
+
+    "validate should return an error when future date is not allowed but the given date is in future" in {
+
+      val date = List(
+        getDate(LocalDateTime.now().plusDays(1)),
+        getDate(LocalDateTime.now().plusMonths(1)),
+        getDate(LocalDateTime.now().plusYears(1))
+      )
+      date.foreach {
+        case (day, month, year) =>
+          DateField.validate(day, month, year, dateField.copy(isFutureDateAllowed = false)) shouldBe Some(s"${dateField.fieldName} date cannot be a future date.")
+      }
+    }
+
+    "validate should return an error when future date is not allowed but the given date is present or past" in {
```

Comment:
Should this be `validate should NOT return an error`?

Reply:
Yeah, good catch. I changed that, thanks.